cryodrgn/types.py

```diff
@@ -0,0 +1,21 @@
+"""Helper module to define cryoDRGN-specific types for static type-checking.
+Usage of this module is recommended for all type annotations instead of direct
+imports of types, to avoid circular import issues.
+
+It is recommended that client code use `import cryodrgn.types` instead of reaching-in for types.
+"""
+from typing import TYPE_CHECKING, Union, List
+import numpy as np
+
+
+if TYPE_CHECKING:  # Set if type-checking
+    # Avoid importing any cryodrgn-specific submodules outside this block.
+    from cryodrgn.mrc import LazyImage
+
+    ImageArray = Union[np.ndarray, List[LazyImage]]
```

Comment:
This is unfortunate... could you just define `ImageArray` where `LazyImage` is defined?

I think we might want to get rid of `ImageArray` and/or `LazyImage` anyway so it might not matter.

Reply:
Before I answer the wrong question, what part did you think is unfortunate? I do think types introduced solely for type-checking should be in a `cryodrgn.types` module (where we check the `TYPE_CHECKING` name) and `cryoDRGN` should `import cryodrgn.types` as needed instead of directly importing names from `cryoDRGN` itself (which will lead to circular import issues for sure).